src/Guesser/TypeGuesserChain.php

```diff
@@ -59,6 +59,9 @@ public function guessType($class, $property, ModelManagerInterface $modelManager
             }
         }
 
-        return TypeGuess::getBestGuess($guesses);
+        $bestGuess = TypeGuess::getBestGuess($guesses);
+        \assert($bestGuess instanceof TypeGuess || null === $bestGuess);
```

Comment:
We could improve the type int of the $guessers property

TypeGuesserInterface[]

With a PR on symfony or phpstan-symfony, we could avoid this assert in the future.

Reply:
@VincentLanglet can you provide POV of the PR?
I don't understand what should `TypeGuess` return in your opinion

Reply:
Im on phone, sorry for the format of my message

In this class we could change

/**
 * @array
 */
 Private $guessers;

Changing array to TypeGuesserInterface[]

In symfony (or phpstan symfony) we could improve the type of the method bestGuess with something like

@phpstan-template T extends Guess
@phpstan-param T[]
@phpstan-return T|null

Reply:
Format is perfectly fine. Thanks 👍

Reply:
I guess @franmomu is already working on it: https://github.com/symfony/symfony/pull/37725